### DISCUSSION TITLE
have a doubt in code of FundMe contract

### DISCUSSION THREAD

**Discussion by karthiksathishjeemain** [Cyfrin/foundry-full-course-cu#991] [Q&A] [answered]
'constructor(address priceFeed) {
        s_priceFeed = AggregatorV3Interface(priceFeed);
        i_owner = msg.sender;
    }'
In this constructor why should we enclose pricefeed within brackets whereas in the PriceConvertor Contract it was just mentioned without assigning any brackets.

library PriceConverter {
    function getPrice(**AggregatorV3Interface priceFeed**) internal view returns (uint256) {
        (, int256 answer,,,) = priceFeed.latestRoundData();
        // ETH/USD rate in 18 digit
        return uint256(answer * 10000000000);
    }

**Comment by hparihar-07** [comment, marked as answer]
Hey @karthiksathishjeemain 
In the getPrice function we are passing priceFeed as an argument to the function with the Aggregatorv3interface as a type but in fundMe contract we are passing priceFeed variable as an argument of type address to the constructor and then passing this stored address in the priceFeed Variable to the Aggregatorv3interface contract ☺️

**Reply by karthiksathishjeemain** [reply]
Thanks for the answer, but I would also like to know 
1. What type of argument does Aggregatorv3interface takes being an interface, I am in a bit of confusion about  "interface" and it's arguments.
2. Also the  Aggregatorv3interface  collects information about the Chainlink PriceFeed Contract even though, it has nothing mentioned as "Chainlink" in the whole interface. Well, what I feel is , the Chainlink PriceFeed Contract is designed in such a way that whenever, it is  called by an interafce like Aggregatorv3interface it gives the details about the contract. Please correct me if I am wrong

**Reply by hparihar-07** [reply]
You can check on chainlink official website they have briefly explained about Aggregatorv3interface
